src/public/app/widgets/containers/ribbon_container.js

```diff
@@ -198,6 +198,10 @@ export default class RibbonContainer extends NoteContextAwareWidget {
                 }
             }
         } else {
+            const activeChild = this.getActiveRibbonWidget();
+            if (activeChild) {
+                activeChild.handleEvent('deactivated', {});
```

Comment:
I am using a new event here since I did not find any existing mechanism to shut down a widget.

Reply:
Have you tried `cleanup()` method in TypeWidget (inherited in NoteMapTypeWidget)? Seems to me it could be used for this. Although it would not work for Ribbon specifically ...

Reply:
It doesn't matter as much for the Ribbon widget, but could still be relevant for cases like #2608